lib/rubocop/cop/flexport/engine_api_boundary.rb

```diff
@@ -179,19 +187,40 @@ def on_const(node)
 
         def on_send(node)
           rails_association_hash_args(node) do |assocation_hash_args|
-            class_name_node = extract_class_name_node(assocation_hash_args)
-            next if class_name_node.nil?
-
-            accessed_engine = extract_model_engine(class_name_node)
-            next if accessed_engine.nil?
-            next if valid_engine_access?(node, accessed_engine)
+            check_for_cross_engine_rails_association(node, assocation_hash_args)
+          end
+          return unless should_check_for_cross_engine_factory_bot?
 
-            add_offense(class_name_node, message: message(accessed_engine))
+          factory_bot_usage(node) do |factory_node|
+            check_for_cross_engine_factory_bot_usage(node, factory_node)
           end
         end
 
+        def check_for_cross_engine_rails_association(node, assocation_hash_args)
+          class_name_node = extract_class_name_node(assocation_hash_args)
+          return if class_name_node.nil?
+
+          accessed_engine = extract_model_engine(class_name_node)
+          return if accessed_engine.nil?
+          return if valid_engine_access?(node, accessed_engine)
+
+          add_offense(class_name_node, message: message(accessed_engine))
+        end
+
+        def check_for_cross_engine_factory_bot_usage(node, factory_node)
+          factory = factory_node.children[0]
+          accessed_engine = factory_engines[factory]
+          return if accessed_engine.nil?
+          return if valid_engine_access?(node, accessed_engine)
```

Comment:
does `allowlisted?` work with this node?

Reply:
in other words, if BarEngine::FooModel model is in _allowlist.rb and I use :foo_model factorybot instantiator from another engine -- does the lint rule complain?

Reply:
No, this doesn't currently respect allowlists. I'll look into extracting the model class name from the factory definitions to enable this.

Reply:
ppl shouldnt be allowlisting models anyway =P

seems fine for now, though please at least indicate this is the scenario in the cop documentation